openroad_interface/validation.py

```diff
@@ -1,85 +1,133 @@
 import shutil
 import os
-import copy
 
 import networkx as nx
-import pandas as pd 
+import pandas as pd
 
-from . import estimation as est
 from . import place_n_route as pnr
 from .var import directory
 from . import def_generator as df
 from . import graph_plotter as gp
 
-def pandas_organize(design_name: str, estimation_dict: dict, detailed_dict: dict ):
-    '''
-    takes data from estimations and detailed calculations and putting them in a pandas dataframe. 
+
+def pandas_organize(design_name: str, estimation_dict: dict, detailed_dict: dict):
+    """
+    takes data from estimations and detailed calculations and putting them in a pandas dataframe.
     param:
         design_name: design name
         estimation_dict: estimated parasitics of each net
-        detailed_dict: detailed parasitics of each net 
-    
+        detailed_dict: detailed parasitics of each net
+
     return:
         result: pandas dataframe
-    '''
+    """
     design_name = design_name.replace(".gml", "")
-    d1 = {"design": [design_name] * len(detailed_dict["res"]), "net_id" : detailed_dict["net"], "method": ["openroad"] * len(detailed_dict["res"]), "res": detailed_dict["res"], "cap": detailed_dict["cap"], "length": detailed_dict["length"]}
+    d1 = {
+        "design": [design_name] * len(detailed_dict["res"]),
+        "net_id": detailed_dict["net"],
+        "method": ["openroad"] * len(detailed_dict["res"]),
+        "res": detailed_dict["res"],
+        "cap": detailed_dict["cap"],
+        "length": detailed_dict["length"],
+    }
     openroad_dataframe = pd.DataFrame(data=d1)
 
-    d2 = {"design": [design_name] * len(estimation_dict["res"]), "net_id" : estimation_dict["net"], "method": ["estimate"] * len(estimation_dict["res"]), "res": estimation_dict["res"], "cap": estimation_dict["cap"], "length": estimation_dict["length"]}
+    d2 = {
+        "design": [design_name] * len(estimation_dict["res"]),
+        "net_id": estimation_dict["net"],
+        "method": ["estimate"] * len(estimation_dict["res"]),
+        "res": estimation_dict["res"],
+        "cap": estimation_dict["cap"],
+        "length": estimation_dict["length"],
+    }
     estimate_dataframe = pd.DataFrame(data=d2)
 
     result = pd.concat([openroad_dataframe, estimate_dataframe])
 
     return result
 
-
-def validation(design_name: str, test_directory: str):
-    '''
-    the main function. generates def file, runs openroad, does all openroad and estimated calculations. 
+def validation(
+    design_name: str,
+    test_directory: str, 
+    arg_parasitics: str
+):
+    graph, net_out_dict, node_output, lef_data, node_to_num = setup(design_name, test_directory)
+    dict, graph = extraction(graph, arg_parasitics, design_name, net_out_dict, node_output, lef_data, node_to_num)
+    return dict, graph
```

Comment:
This should be a function in pnr. This is the function we want to call from our simulation.
It should not be called validation. call it `pnr`